fairlearn/reductions/grid_search/grid_search.py

```diff
@@ -52,31 +63,43 @@ def fit(self, X, Y, **kwargs):
         if self._KW_NUMBER_LAGRANGE_MULTIPLIERS in kwargs:
             number_of_lagrange_multipliers = kwargs[self._KW_NUMBER_LAGRANGE_MULTIPLIERS]
 
+        # Extract the target attribute
+        A = self._make_vector(aux_data, "aux_data")
+
+        # Extract the Y values
+        Y_vector = self._make_vector(Y, "Y")
+
+        X_rows, _ = self._get_matrix_shape(X, "X")
+        if X_rows != Y_vector.shape[0]:
+            raise RuntimeError(self._MESSAGE_X_Y_ROWS)
+        if X_rows != A.shape[0]:
+            raise RuntimeError(self._MESSAGE_X_A_ROWS)
+
         # Prep the quality metric
-        self.quality_metric.set_data(X, Y, aux_data)
+        self.quality_metric.set_data(X, Y_vector, A)
```

Comment:
Shouldn't we use lower case y for vectors?

Reply:
True. I've just updated a bunch of these.